docusaurus/docs/Android/03-ui/04-components/03-message-list.md

```diff
@@ -24,62 +77,113 @@
 | --- | --- |
 |![Message_options_in light mode](../../assets/message_options_light.png)|![Message options in dark mode](../../assets/message_options_dark.png)|
 
-Most of the actions work out of the box but you can change their behavior using different listeners and handlers:
+If you're not going to use out of the box `MessageListViewModel` with it's handlers or just want to override action handlers you should define these handlers:
+```kotlin
+fun setActionHandlers() {
+        messageListView.setLastMessageReadHandler {
+            // Handle when last message got read
+        }
+        messageListView.setEndRegionReachedHandler {
+            // Handle when end region reached
+        }
+        messageListView.setMessageDeleteHandler { message: Message ->
+            // Handle when message is going to be deleted
+        }
+        messageListView.setThreadStartHandler { message: Message ->
+            // Handle when new thread for message is started
+        }
+        messageListView.setMessageFlagHandler { message: Message ->
+           // Handle when message is going to be flagged
+        }
+        messageListView.setGiphySendHandler { message: Message, giphyAction: GiphyAction ->
+            // Handle when some giphyAction is going to be performed
+        }
+        messageListView.setMessageRetryHandler { message: Message ->
+            // Handle when some failed message is going to be retried
+        }
+        messageListView.setMessageReactionHandler { message: Message, reactionType: String ->
+            // Handle when some reaction for message is going to be send
+        }
+        messageListView.setUserMuteHandler { user: User ->
+            // Handle when a user is going to be muted
+        }
+        messageListView.setUserUnmuteHandler { user: User ->
+            // Handle when a user is going to be unmuted
+        }
+        messageListView.setUserBlockHandler { user: User, cid: String ->
+            // Handle when a user is going to be blocked in the channel with cid
+        }
+        messageListView.setMessageReplyHandler { cid: String, message: Message ->
+            // Handle when message is going to be replied in the channel with cid
+        }
+        messageListView.setAttachmentDownloadHandler { attachment: Attachment ->
+            // Handle when attachment is going to be downloaded
+        }
+    }
+``` 
+
+---
+**NOTE**
+
+Handlers must be set before passing any data to `MessageListView`. So if you don't use default binding, please, make sure you defined them.
+
+___
+
+### Listeners
+
+Except of required handlers you're able to set listeners to get events when something happens:
```

Comment:
It might be worth it writing that the same as with Handlers - default ones are set when binding view model with view